code/game/objects/effects/aliens.dm

```diff
@@ -153,6 +153,8 @@
 		var/mob/living/carbon/xenomorph/X = AM
 		if (X.hivenumber != hivenumber)
 			apply_spray(AM)
+		else
+			X.ExtinguishMob()
```

Comment:
The acid spray extinguishes any `/living` subtype if it hits someone in `Initialize()`, so it would make sense for this to do the same. (Currently it'll only extinguish xenomorphs)
https://github.com/cmss13-devs/cmss13/blob/b5e180c53acbca0fcc195e9b9d13a7c42b3f4a92/code/game/objects/effects/aliens.dm#L107-L109

Looking into it a bit, this proc could actually be refactored and shortened quite a bit since the `ally_of_hivenumber()` proc is on `/living`, and contains a `xeno.hivenumber != hivenumber` check for xenos as well.
Something like this would work:
```dm
/obj/effect/xenomorph/spray/Crossed(AM as mob|obj)
	..()
	if(isliving(AM))
		var/mob/living/living_mob = AM
		if(living_mob.ally_of_hivenumber(hivenumber))
			living_mob.ExtinguishMob()
		else
			apply_spray(living_mob)
	else if(isVehicleMultitile(AM))
		var/obj/vehicle/multitile/V = AM
		V.handle_acidic_environment(src)
```

Reply:
Good point